rust/lance-index/src/vector/ivf.rs

```diff
@@ -446,7 +472,34 @@ impl<T: ArrowFloatType + Dot + L2 + Cosine + 'static> Ivf for IvfImpl<T> {
             ),
             location: location!(),
         })?;
-        let part_ids = self.compute_partitions(data).await?;
+
+        let part_ids = match (&self.precomputed_partitions, batch.column_by_name(ROW_ID)) {
+            (Some(partitions), Some(row_ids)) => {
+                debug!("Using precomputed partitions for partitions");
+                let mut builder = UInt32Builder::new();
+                for row in row_ids
+                    .as_any()
+                    .downcast_ref::<UInt64Array>()
```

Comment:
Minor nit: if you're just going to unwrap anyways you can use `row_ids.as_primitive::<UInt64Type>()`